src/api.js

```diff
@@ -64,11 +59,10 @@ class Api {
 					if ( err ) {
 						return reject( err );
 					}
-					if ( sets.length ) {
+					if ( rows !== undefined ) {
 						sets.push( rows );
-						return resolve( sets );
 					}
-					return resolve( rows );
+					return resolve( sets );
```

Comment:
I love the distinction between `query` and `querySets`, but I still feel like this is more complicated than it needs to be. Wouldn't this work?
```
			return new Promise( ( resolve, reject ) => {
				const request = new Request( _sql, err => {
					if ( err ) {
						return reject( err );
					}
					return resolve( sets );
				} );
				addRequestParams( request, params );
				request.on( "columnMetadata", () => {
					rows = [];
					sets.push( rows );
				} );
				request.on( "row", obj => rows.push( transformRow( obj ) ) );
				conn.execSql( request );
			} );
```

Reply:
@calvinb That looks much better. I'll give that a go!